cli/tests/pcluster/validators/test_secret_validators.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from pcluster.aws.common import AWSClientError
+from pcluster.validators.common import FailureLevel
+from pcluster.validators.secret_validators import MungeKeySecretArnValidator
+from tests.pcluster.validators.utils import assert_failure_level, assert_failure_messages
+
+
+@pytest.mark.parametrize(
+    "munge_key_secret_arn, region, mock_response, expected_response, error_from_aws_service, expected_failure_level",
+    [
+        (
+            "arn:aws:secretsmanager:us-west-2:123456789012:secret:testSecret",
+            "us-west-2",
+            {"SecretString": "validBase64ValueMakeItInRange[256,8192]MakeItInRange[256,8192]"},
+            "",
+            None,
+            None,
+        ),
+        (
+            "arn:aws:secretsmanager:us-west-2:123456789012:secret:testSecret",
+            "us-west-2",
+            {"SecretString": "validBase64Value"},
+            "The size of the decoded munge key in the secret "
+            "arn:aws:secretsmanager:us-west-2:123456789012:secret:testSecret is 96 bits "
+            "which is outside the allowed range [256-8192].",
+            None,
+            FailureLevel.ERROR,
+        ),
+        (
+            "arn:aws:otherService:us-west-2:123456789012:secret:testSecret",
+            "us-west-2",
+            {"SecretString": "validBase64ValueMakeItInRange[256,8192]MakeItInRange[256,8192]"},
+            "The secret arn:aws:otherService:us-west-2:123456789012:secret:testSecret is not supported "
+            "in region us-west-2.",
+            None,
+            FailureLevel.ERROR,
+        ),
+        (
+            "arn:aws:secretsmanager:us-west-2:123456789012:otherResource:testSecret",
+            "us-west-2",
+            {"SecretString": "validBase64ValueMakeItInRange[256,8192]MakeItInRange[256,8192]"},
+            "The secret arn:aws:secretsmanager:us-west-2:123456789012:otherResource:testSecret is not supported"
+            " in region us-west-2.",
+            None,
+            FailureLevel.ERROR,
+        ),
+        (
+            "arn:aws:secretsmanager:us-west-2:123456789012:secret:testSecret",
+            "us-west-2",
+            {"SecretString": "invalidBase64MakeItInRange[256,8192]MakeItInRange[256,8192]"},
```

Comment:
Please put a comment about why "valid..." is a valid encoded secret and "invalid..." is an invalid one (and this has to do with the number of characters in the encoded string, including the number of padding chars).

Reply:
Done!